qds_sdk/cluster.py

```diff
@@ -1168,6 +1214,217 @@ def minimal_payload(self):
+        self.internal['spark_s3_package_name'] = spark_s3_package_name
+        self.internal['zeppelin_s3_package_name'] = zeppelin_s3_package_name
+        self.internal['zeppelin_interpreter_mode'] = zeppelin_interpreter_mode
+        self.internal['tunnel_server_ip'] = tunnel_server_ip
+        self.internal['restrict_ssh_access'] = restrict_ssh_access
+        self.internal['ami_overrides'] = ami_overrides
+        self.internal['image_uri_overrides'] = image_uri_overrides
+
+    def set_cluster_information(self, master_instance_type=None, slave_instance_type=None, min_nodes=None,
+                                max_nodes=None, cluster_name=None, node_bootstrap=None,
+                                disallow_cluster_termination=None, force_tunnel=None, fallback_to_ondemand=None,
+                                customer_ssh_key=None, custom_tags=None, size=None, count=None, type=None,
+                                upscaling_config=None, encryption=None, heterogeneous_config=None,
+                                slave_request_type=None, maximum_bid_price_percentage=None,
+                                timeout_for_request=None, maximum_spot_instance_percentage=None,
+                                stable_maximum_bid_price_percentage=None, stable_timeout_for_request=None,
+                                               stable_allow_fallback=None):
+        self.cluster_info['master_instance_type'] = master_instance_type
+        self.cluster_info['slave_instance_type']  = slave_instance_type
+        self.cluster_info['min_nodes'] = min_nodes
+        self.cluster_info['max_nodes'] = max_nodes
+        self.cluster_info['cluster_name'] = cluster_name
+        self.cluster_info['node_bootstrap'] = node_bootstrap
+        self.cluster_info['disallow_cluster_termination'] = disallow_cluster_termination
+        self.cluster_info['force_tunnel'] = force_tunnel
+        self.cluster_info['fallback_to_ondemand'] = fallback_to_ondemand
+        self.cluster_info['customer_ssh_key'] = customer_ssh_key
+        if custom_tags and custom_tags.strip():
+            try:
+                self.cluster_info['custom_tags'] = json.loads(custom_tags.strip())
+            except Exception as e:
+                raise Exception("Invalid JSON string for custom ec2 tags: %s" % e.message)
+        self.set_data_disk(size, count, type, upscaling_config, encryption)
+        self.cluster_info['heterogeneous_config'] = heterogeneous_config
+        self.cluster_info['slave_request_type'] = slave_request_type
+        self.set_spot_settings(maximum_bid_price_percentage, timeout_for_request,
+                                        maximum_spot_instance_percentage, stable_maximum_bid_price_percentage,
+                                        stable_timeout_for_request, stable_allow_fallback)
+
+    def set_spot_settings(self, maximum_bid_price_percentage=None, timeout_for_request=None,
```

Comment:
If I call `set_spot_settings(..)` followed by `set_cluster_information(..)` without the spot settings specified. We'll end up overwriting, correct? Is that desirable? Similar problems will happen when you make a member method of this class implicitly from within another member method that takes a super set of arguments.